iox_catalog/src/postgres.rs

```diff
@@ -240,6 +217,143 @@ impl Catalog for PostgresCatalog {
     }
 }
 
+/// Creates a new [`sqlx::Pool`] from a database config and an explicit DSN.
+///
+/// This function doesn't support the IDPE specific `dsn-file://` uri scheme.
+async fn new_raw_pool(
+    app_name: &str,
+    schema_name: &str,
```

Comment:
As you're already creating `Arc<str>` in the caller, you can use arcs here and avoid some clones:

```suggestion
    app_name: Arc<str>,
    schema_name: Arc<str>,
```

It's not a big deal though - it's 1 clone of each str per connection per `new_raw_pool()` call.